File: app/src/main/java/fr/xgouchet/chronorg/ui/fragments/PortalListFragment.java
package fr.xgouchet.chronorg.ui.fragments;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import fr.xgouchet.chronorg.data.formatters.ReadablePeriodFormatter;
import fr.xgouchet.chronorg.data.models.Portal;
import fr.xgouchet.chronorg.ui.activities.PortalEditActivity;
import fr.xgouchet.chronorg.ui.adapters.PortalsAdapter;

/**
 * @author Xavier Gouchet
 */
public class PortalListFragment extends BaseListFragment<Portal, PortalsAdapter> {

    private static final String ARGUMENT_PROJECT_ID = "project_id";

    private int projectId = -1;

    public static PortalListFragment createFragment(int projectId) {
        PortalListFragment fragment = new PortalListFragment();

        Bundle args = new Bundle(1);
        args.putInt(ARGUMENT_PROJECT_ID, projectId);
        fragment.setArguments(args);
        return fragment;
    }

    @Override protected PortalsAdapter getAdapter() {
        // TODO inject formatter
        return new PortalsAdapter(this, new ReadablePeriodFormatter());
    }

    @Override public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle args = getArguments();
        if ((args != null) && args.containsKey(ARGUMENT_PROJECT_ID)) {
            projectId = args.getInt(ARGUMENT_PROJECT_ID);
        }
    }

    @Override public void showCreateItemUi() {
        Intent intent = PortalEditActivity.intentNewPortal(getActivity(), projectId);
        getActivity().startActivity(intent);
    }

    @Override public void showItem(@NonNull Portal item) {
        Intent intent = PortalEditActivity.intentEditPortal(getActivity(), item);
        getActivity().startActivity(intent);
    }

}

File: khronorg/src/main/java/fr/xgouchet/khronorg/feature/portals/PortalListFragment.kt
package fr.xgouchet.khronorg.feature.portals

import android.os.Bundle
import android.view.View
import fr.xgouchet.khronorg.commons.formatters.DefaultIntervalFormatter
import fr.xgouchet.khronorg.ui.adapters.BaseAdapter
import fr.xgouchet.khronorg.ui.fragments.ListFragment

/**
 * @author Xavier F. Gouchet
 */
class PortalListFragment : ListFragment<Portal>(true) {


    override val adapter: BaseAdapter<Portal> = PortalAdapter(DefaultIntervalFormatter, this)

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        fab.setOnClickListener { e -> presenter.itemSelected(null) }
    }
}